client/src/components/HomePage/index.tsx

```diff
@@ -50,7 +50,7 @@ export function HomePage() {
         // Prevent the page from refreshing
         e.preventDefault();
       
-        if (pUsername.length === 0) {
+        if (pUsername.trim().length === 0) {
```

Comment:
should trim payload before sending to server, otherwise you could have "alex" and "alex " and "alex   "